src/uploadGraphqlSchemaCommand.ts

```diff
@@ -0,0 +1,52 @@
+import vscode from 'vscode';
+
+export default (
+  mode: string = 'merge',
+  adminSecretKey: string,
+  outputChannel: vscode.OutputChannel
+) => async () => {
+  const { activeTextEditor } = vscode.window;
+
+  if (!activeTextEditor || activeTextEditor.document.languageId !== 'graphql') {
+    vscode.window.showWarningMessage(
+      'You have to select a Graphql document to upload a schema.'
+    );
+    return;
+  }
+
+
+  const selection = activeTextEditor.selection;
+  const selectedText = activeTextEditor.document.getText(selection);
+  const fqlExpression = selectedText.length > 0 ? selectedText : activeTextEditor.document.getText();
+  if (fqlExpression.length < 1) {
+     vscode.window.showWarningMessage(
+      'Selected file or selected text must have a GraphQL Schema to run'
+    );
+    
+    return;
+  }
+
+  outputChannel.appendLine('');
+  outputChannel.appendLine(`UPLOADING SCHEMA (mode=${mode}): ${activeTextEditor.document.fileName}`);
+  outputChannel.show();
+
+  try {
+    const buffer = Buffer.from(fqlExpression, "utf-8");
+    const result = await fetch(`https://graphql.fauna.com/import?mode=${mode}`, {
+      method: "POST",
+      headers: {
+        'AUTHORIZATION': `Bearer ${adminSecretKey}`
+      },
+
+      body: buffer
+    });
+    outputChannel.appendLine('');
+    outputChannel.appendLine('RESPONSE:');
+    outputChannel.appendLine(await result.text());
```

Comment:
@BrunoQuaresma Should we be calling `.json()` here instead?

Reply:
Part of this result logs out the API key. Do we explicitly want to do that?

Reply:
The response from the endpoint is not a JSON object, but text. Calling `.json()` would probably fail. https://docs.fauna.com/fauna/current/api/graphql/endpoints#errors (link shows that return type of /import endpoint is plaintext.

Reply:
> Part of this result logs out the API key. Do we explicitly want to do that?

I was wondering about this myself. The response does include it explicitly. We could strip it off before logging. But since that same key is already visible to the user in their settings, I don't know if its a huge problem.

Reply:
> The response from the endpoint is not a JSON object, but text. Calling `.json()` would probably fail. https://docs.fauna.com/fauna/current/api/graphql/endpoints#errors (link shows that return type of /import endpoint is plaintext.

@nksaraf Thanks for the clarification. I wasn't 100% sure what was coming back, so just asking for my own clarification 🙏

Reply:
> > Part of this result logs out the API key. Do we explicitly want to do that?
> 
> I was wondering about this myself. The response does include it explicitly. We could strip it off before logging. But since that same key is already visible to the user in their settings, I don't know if its a huge problem.

@nksaraf Agreed. Since the user already has entered the key in their config file, I don't think it's a huge deal. Displaying it on the screen may be a security concern if they're working in an open environment 🤷‍♂ 

@BrunoQuaresma @danieltodonnell @fauna-rodrigo What do you think?